src/commands/aksClusterProperties/aksClusterProperties.ts

```diff
@@ -19,37 +19,112 @@ export default async function aksClusterProperties(
       return;
-    }
+  }
 
-    await loadClusterProperties(cluster.result, extensionPath.result);
+  await loadWebViewClusterProperties(cloudTarget, clusterData.result, clusterState);
 }
 
-async function loadClusterProperties(
+async function loadWebViewClusterProperties(
     cloudTarget: AksClusterTreeItem,
-    extensionPath: string) {
+    clusterInfo: ClusterARMResponse,
+    clusterState: string
+) {
 
     const clustername = cloudTarget.name;
-    await longRunning(`Loading ${clustername} cluster properties.`,
-      async () => {
-        const clusterInfo = await getClusterProperties(cloudTarget, clustername);
-        if (failed(clusterInfo)) {
-          vscode.window.showErrorMessage(clusterInfo.error);
-          return;
-        }
 
+    const extensionPath = getExtensionPath();
+    if (failed(extensionPath)) {
+      vscode.window.showErrorMessage(extensionPath.error);
+      return;
+    }
+
+    await longRunning(`Loading webview ${clustername} for cluster properties.`,
+      async () => {
+        let clusterData: ClusterARMResponse | undefined;
         const webview = createWebView('AKS Cluster Properties', `AKS properties view for: ${clustername}`);
 
-        webview.html = getWebviewContent(clusterInfo.result, extensionPath);
+        webview.onDidReceiveMessage(
+          async (message) => {
+              if (message.command === "startCluster") {
+                clusterData = await onRecievePerformOperations(cloudTarget, clustername, 'start');
+              } else if (message.command === "stopCluster") {
+                clusterData = await onRecievePerformOperations(cloudTarget, clustername, 'stop');
+              }
+              const clusterState = await determineClusterState(cloudTarget, clustername);
+              if (!clusterState || !clusterData) {
+                  return;
+              }
+              webview.html = getWebviewContent(clusterData, clusterState, extensionPath.result);
+          },
+          undefined
+      );
+
+        webview.html = getWebviewContent(clusterInfo, clusterState, extensionPath.result);
       }
     );
 }
 
+async function getClusterData(
+  cloudTarget: AksClusterTreeItem
+): Promise<Errorable<ClusterARMResponse>> {
+
+    return await longRunning(`Loading ${cloudTarget.name} cluster data.`,
+      async () => {
+        const clusterInfo = await getClusterProperties(cloudTarget, cloudTarget.name);
+        return clusterInfo;
+      }
+    );
+}
+
+async function onRecievePerformOperations(
+  cloudTarget: AksClusterTreeItem,
+  clusterName: string,
+  eventName: string
+): Promise<ClusterARMResponse | undefined> {
+
+    let startStopClusterInfo: Errorable<string>;
+    if (eventName === 'start') {
+     startStopClusterInfo = await longRunning(`Starting cluster.`, () => startCluster(cloudTarget, clusterName) );
+    } else {
+      startStopClusterInfo = await longRunning(`Stopping cluster.`, () => stopCluster(cloudTarget, clusterName));
```

Comment:
Just to protect against weird bugs in the future I'd prefer to use an explicit `case` (or `else if`) for `stop`, rather than the catch-all `else`. And then the `default`/`else` case would be reserved for reporting an error that should never happen.